### DISCUSSION TITLE
Unable to run NativeModuleSample

### DISCUSSION THREAD

**Discussion by treadways** [microsoft/react-native-windows#11183] [Q&A] [answered]
After having trouble implementing a Native Module for Windows following the documentation on Microsoft's website I tried running the NativeModuleSample\cppwinrt. After following the instructions in the respective README.md I am left with the following error:
![screenshot_bug](https://user-images.githubusercontent.com/33097069/217646146-23319e8a-5ab5-46a0-961d-e388f213a1f2.PNG)

Any idea what I am doing wrong? Also is the documentation on how to create a native module correct on Microsoft's website? I was able to easily create a Native Module for android but am having continuous issues doing so for the Windows version.

**Comment by ghost** [comment]
@treadways run npx react-native run-windows --logging so we can get details of the problem

**Reply by treadways** [reply]
@Victor0814gui  I added a screenshot of running the same command with --logging. Let me know if you see anything. Thanks!

**Comment by treadways** [comment]
![image](https://user-images.githubusercontent.com/33097069/218510763-6689ae98-062f-4be9-a2df-46d00b928e6c.png)
![image](https://user-images.githubusercontent.com/33097069/218510859-aab87f92-86f8-4340-9edd-4169ff7d8a08.png)
![image](https://user-images.githubusercontent.com/33097069/218510954-fc1a0595-c312-4cd8-a10b-21166fe65118.png)
![image](https://user-images.githubusercontent.com/33097069/218511036-772cc641-9734-435d-8c59-555b46c10d44.png)
![image](https://user-images.githubusercontent.com/33097069/218511137-9d6ad2f6-90a5-4649-93b0-a7e4274a29e0.png)
![image](https://user-images.githubusercontent.com/33097069/218511215-9a4d36ac-425f-4d01-a74e-045452dd59bb.png)
![image](https://user-images.githubusercontent.com/33097069/218511301-ae68ec64-cc1b-425e-b8ea-4e6b90b94c61.png)
![image](https://user-images.githubusercontent.com/33097069/218511390-58159e8f-948a-4ef3-ba86-8644b3c64662.png)
![image](https://user-images.githubusercontent.com/33097069/218511541-0d7d3d24-6511-4d27-9d67-87693cc3f4aa.png)
![image](https://user-images.githubusercontent.com/33097069/218511611-53f41de3-c6d5-4c50-b1bd-4cc0fad0a4d5.png)
![image](https://user-images.githubusercontent.com/33097069/218511710-9308ff19-f39a-464b-bd89-6ba7fcb4c2c1.png)
![image](https://user-images.githubusercontent.com/33097069/218511806-63686a8d-df48-4cbd-9c7d-ecb3cf5173ea.png)
![image](https://user-images.githubusercontent.com/33097069/218511923-2c99445c-2d70-4e6b-8c55-2d7a97bb1ba2.png)
![image](https://user-images.githubusercontent.com/33097069/218512025-0738606b-7f6e-4d1d-81f5-335fcef77993.png)

**Reply by chrisglein** [reply]
An internal MIDL error? That's odd.
In our bug report template we have more information we'd gather to help figure out what's going on. Can you run this command and share the output?
`npx react-native info`

**Reply by treadways** [reply]
![image](https://user-images.githubusercontent.com/33097069/221676082-a994b350-67e8-4bae-9843-903f938e8e33.png)

**Reply by chrisglein** [reply]
MIDL is a build tool that's a bit beyond React Native's role in this. It's used to compile UWP apps and such and is an independent tool. An error that indicates... something deeper probably? A next step would be to try building this in Visual Studio instead of via the commandline, and enabling verbose build logging. There's a link in your error message there that indicates some troubleshooting information - not sure if it's useful or not.

Example of other issue that involved tangoing with MIDL: https://github.com/microsoft/react-native-windows/issues/10414

**Comment by ghost** [comment]
@treadways  have you tried using visual sutudio 2019 or the 2022 to build the solution? lately I have had problems with the CLI related to visual studio 2022, but also related to 2019, so what I do is open the solution directly in visual studio and build the solution, as it is written in the documentation.

**Reply by treadways** [reply]
I tried both and am still having errors

**Reply by treadways** [reply]
![image](https://user-images.githubusercontent.com/33097069/234989334-821cd0ca-2cd5-45fe-8c9a-b11057cfd035.png)
![image](https://user-images.githubusercontent.com/33097069/234989401-c46e30b8-d6b6-4cd5-b1d9-96f29fc9829a.png)

**Reply by ghost** [reply]
@treadways  it looks like you used the react-native cli to build the solution and opened the project folder instead of the solution, if you did look for a file in "windows/project-name/project-name.sln" and click on the button with the debug option selected.

**Reply by chrisglein** [reply, marked as answer]
Replicated your error following the instructions. Opened a bug: https://github.com/microsoft/react-native-windows-samples/issues/832

Has the text version of the error above. In the future, if you can copy out the text instead of providing only a screenshot, that's helpful.
